pyomo/util/latex_printer.py

```diff
@@ -0,0 +1,619 @@
+import pyomo.environ as pe
+from pyomo.core.expr.visitor import StreamBasedExpressionVisitor
+from pyomo.core.expr import (
+    NegationExpression,
+    ProductExpression,
+    DivisionExpression,
+    PowExpression,
+    AbsExpression,
+    UnaryFunctionExpression,
+    MonomialTermExpression,
+    LinearExpression,
+    SumExpression,
+    EqualityExpression,
+    InequalityExpression,
+    RangedExpression,
+    Expr_ifExpression,
+    ExternalFunctionExpression,
+)
+
+from pyomo.core.expr.base import ExpressionBase
+from pyomo.core.base.expression import ScalarExpression, _GeneralExpressionData
+from pyomo.core.base.objective import ScalarObjective, _GeneralObjectiveData
+import pyomo.core.kernel as kernel
+from pyomo.core.expr.template_expr import (
+    GetItemExpression,
+    GetAttrExpression,
+    TemplateSumExpression,
+    IndexTemplate,
+)
+from pyomo.core.expr.template_expr import Numeric_GetItemExpression
+from pyomo.core.expr.template_expr import templatize_constraint, resolve_template
+from pyomo.core.base.var import ScalarVar, _GeneralVarData, IndexedVar
+from pyomo.core.base.constraint import ScalarConstraint, IndexedConstraint
+from pyomo.core.expr.template_expr import templatize_rule
+
+from pyomo.core.base.external import _PythonCallbackFunctionID
+
+from pyomo.core.base.block import _BlockData
+
+from pyomo.repn.util import ExprType
+
+_CONSTANT = ExprType.CONSTANT
+_MONOMIAL = ExprType.MONOMIAL
+_GENERAL = ExprType.GENERAL
+
+# see:  https://github.com/Pyomo/pyomo/blob/main/pyomo/repn/plugins/nl_writer.py
+
+
+def templatize_expression(expr):
+    expr, indices = templatize_rule(expr.parent_block(), expr._rule, expr.index_set())
+    return (expr, indices)
+
+
+def templatize_passthrough(con):
+    return (con, [])
+
+
+def handle_negation_node(node, arg1):
+    childPrecedence = []
```

Comment:
This is a general quibble for the whole file: it's PEP8-standard to use snake case (`child_precedence`) for variables: https://peps.python.org/pep-0008/#function-and-variable-names

Reply:
No

Reply:
Seriously though, I use camel case a lot because the _ character causes additional stress on my hands while typing and so I strongly prefer not to use it as much as possible. I do sometimes use it when I need a clear separation in my variable names, but otherwise, I strongly prefer not to do this